allennlp/common/from_params.py

```diff
@@ -241,6 +243,10 @@ def from_params(cls: Type[T], params: Params, **extras) -> T:
         # pylint: disable=protected-access
         from allennlp.common.registrable import Registrable  # import here to avoid circular imports
 
+        module = cls.from_pretrained_params(params) # type: ignore
```

Comment:
I don't quite understand your suggestion, what would `_load_module` do, and how would classes with custom `from_params` methods call it?

Reply:
Maybe it's a utility function instead of a private method (I was imagining it would be static).  The method just literally has the loading logic that's in lines 344-363, without the initial checks for `None`.  Doing it this way just means that you don't need an extra method that you're calling on `cls`, which is a pretty unfortunate requirement (as you can see in the other files modified in this PR).

Reply:
I still don't see how you'd avoid it.

e.g. `FeedForward.from_params` is right now its own method with custom logic. how would you get that method to call the utility function, other than explicitly doing so?

Reply:
I think explicit calling would have to be done. @matt-gardner  Yes, if I understand you right, I had done the same before last commit having plain utility function `from_pretrained_params` in the from_params file.

Reply:
Oh, I think I see the problem.  I was imagining the `Model` class noticing that it needed a `FeedForward`, seeing that the params specified a pretrained `FeedForward`, and instantiating it separately instead of recursing on `FeedForward.from_params`.  The logic for "instantiating it separately" is still that utility method, but you're right that it needs to be a little different in `from_params` - you need to do some lookahead instead of putting it at the top like this.  

Sorry, being brief, because I'm running late already.  If this doesn't make sense, I can clarify more later today.

Reply:
Btw, if we do so, we would also need to pass the class `cls` explicitly. Eg. from_pretrained_params(cls, params) in feedforward's from_params.